modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceConfigurationManager.java

```diff
@@ -257,40 +258,111 @@ private void validate(ServiceTemplateEntity serviceTemplateEntity,
+            if (Objects.isNull(oldestRequest)) {
+                return ResponseEntity
+                        .status(HttpStatus.NO_CONTENT)
+                        .body(new ServiceConfigurationChangeRequest());
+            }
+            List<DeployResource> deployResources =
+                    getDeployResources(UUID.fromString(serviceId), DeployResourceKind.VM);
+            if (Objects.isNull(oldestRequest.getResourceName())) {
+                validateConfigManager(serviceId, oldestRequest.getConfigManager(),
+                        resourceName, deployResources);
+            }
+            ServiceConfigurationUpdateRequest request =
+                    updateServiceConfigurationUpdateRequest(oldestRequest, resourceName);
+            if (Objects.isNull(request)) {
+                return ResponseEntity
+                        .status(HttpStatus.NO_CONTENT)
+                        .body(new ServiceConfigurationChangeRequest());
+            }
+            return ResponseEntity
+                    .status(HttpStatus.OK)
+                    .body(getServiceConfigurationChangeRequest(request, deployResources));
+        } catch (LockTimeoutException e) {
+            log.error("Update service configuration update request object timed out", e);
+            return ResponseEntity
+                    .status(HttpStatus.NO_CONTENT)
+                    .body(new ServiceConfigurationChangeRequest());
+        }
+    }
+
+    private ServiceConfigurationUpdateRequest updateServiceConfigurationUpdateRequest(
+            ServiceConfigurationUpdateRequest request, String resourceName) {
+        if (Objects.isNull(request.getResourceName())) {
+            request.setResourceName(resourceName);
+        }
+        request.setStatus(ServiceConfigurationStatus.PROCESSING);
+        return serviceConfigurationUpdateStorage.storeAndFlush(request);
+    }
+
+    private ServiceConfigurationUpdateRequest
+            getOldestServiceConfigurationChangeRequest(String serviceId, String resourceName) {
         ServiceConfigurationUpdateRequestQueryModel model =
                 new ServiceConfigurationUpdateRequestQueryModel(null, UUID.fromString(serviceId),
-                        resourceName, null, ServiceConfigurationStatus.PENDING);
+                        null, null, ServiceConfigurationStatus.PENDING);
         List<ServiceConfigurationUpdateRequest> requests =
                 serviceConfigurationUpdateStorage.listServiceConfigurationUpdateRequests(model);
         if (CollectionUtils.isEmpty(requests)) {
-            return ResponseEntity.status(HttpStatus.NO_CONTENT)
-                    .body(new ServiceConfigurationChangeRequest());
+            return null;
         }
-        Optional<ServiceConfigurationUpdateRequest> latestRequest = requests.stream()
+        Optional<ServiceConfigurationUpdateRequest> oldestRequestOptional = requests.stream()
+                .filter(request -> Objects.isNull(request.getResourceName())
```

Comment:
here we check if its null, but we must also check if the configManager is same  as the group to which the resource_name in the request belongs to.